### PR TITLE
Drop UIA property change events in 1Password to fix poor performance.

### LINKED ISSUE #10508: NVDA very sluggish when navigating 1password for windows desktop
### Steps to reproduce:
1. Download the latest windows desktop version of 1password from http://www.1password.com
2. If you don't have a 1password account, sign up for a free trial.
3. Add a few passwords to your vault.
4. Tab to the list of passwords and use the arrow keys to navigate. NVDA will pause for around 1.5-2 seconds as you press the arrow keys to navigate.
JAWS and Narrator don't present this issue.

### Actual behavior:
NVDA very sluggish when navigating between items in password list. No errors show up in the log.

### Expected behavior:
NVDA should be as responsive as in other list views.

#### NVDA installed/portable/running from source:
NVDA installed 

#### NVDA version:
. 2019.2.1.
#### Windows version:
insider build 19023.1
#### Name and version of other software in use when reproducing the issue:
1password desktop version 7.3.712
#### Does the issue still occur after restarting your PC?
yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
same behavior on alpha-19195,8306ec15
[nvda.log](https://github.com/nvaccess/nvda/files/3863695/nvda.log)

### PR DESCRIPTION
**Pull request by jcsteh**
### Link to issue number:
Fixes #10508.

### Summary of the issue:
When moving between items in the list in 1Password, response is extremely sluggish, frequently taking over a second to report the newly selected item.

### Description of how this pull request fixes the issue:
1Password floods property change events whenever you move between items in the list. Because NVDA listens to all property changes and instantiates NVDAObjects in most cases, this causes severe sluggishness.

To work around this, drop all property change events from 1Password as early as possible. AppModule.shouldProcessUIAPropertyChangedEvent has been introduced to facilitate this.

### Testing performed:
Moved through items in the 1Password list with the down arrow key. Observed that response is much faster.

### Known issues with pull request:
This is somewhat of a hack. 1Password shouldn't be flooding these events. That said, Narrator and JAWS aren't impacted like NVDA is and there are certainly improvements that could be made to NVDA's UIA event handling. For now, this is the most effective solution without significant refactor/re-architecture of UIA event handling.

### Change log entry:

Bug fixes:
`- NVDA is no longer extremely sluggish when moving within the list of items in 1Password. (#10508)`

### CODE REVIEW

**Comment by jcsteh** [comment]
Updated to only reject property changes that are actually flooded (name, item status, enabled).

**Comment by francipvb** [comment]
Hello,


I've proposed some sort of UIA NVDA object caching mechanism to attack
this issue, because UIA fires too much events in complex applications
like Visual Studio.


Cheers,

-- 
Cuando tus fuerzas terminan, las de mi Dios comienzan.

**Review by feerrenrut — APPROVED** [review]
This looks fine to me. @jcsteh Would you like me to wait for a review from @michaelDCurran  before merging?

**Comment by jcsteh** [comment]
Thanks, @feerrenrut! Please go ahead and merge.

File: source/appModuleHandler.py
Comment (derekriemer) [inline source/appModuleHandler.py:496]: can you briefly describe what these params mean so I can send a followup to document them?

**Comment by jcsteh** [comment]
Sorry. I figured anyone wanting to mess with UIA events at this low level would already understand what these were based on the UIA interfaces, so I didn't see the need to document them. However, I probably should've documented them regardless. :) sender is the raw UIA element sending the event. propertyId is the raw UIA property id for the property which changed.

**Pull request merged**
